walletconnect-wallet-standard/src/index.js

```diff
@@ -73,265 +63,477 @@ const getConnectParams = (chains) => ({
+    const topics = Object.values(this.#topics);
+    const noChange = topics.every((opt) => opt === null);
+    this.#topics = {
+      "solana:mainnet": null,
+      "solana:devnet": null,
+    };
+    this.#accounts = {
+      "solana:mainnet": [],
+      "solana:devnet": [],
+    };
+
+    if (!noChange) {
+      this.#emit("change", { accounts: this.accounts });
+      for (const topic of topics) {
+        if (topic === null) {
+          continue;
         }
-        throw new ClientNotInitializedError();
-      },
-    },
+        // eslint-disable-next-line no-await-in-loop
+        await this.#client.disconnect({
+          topic,
+          reason: getSdkError("USER_DISCONNECTED"),
+        });
+      }
+    }
+  }
+
+  /** @type {import("@wallet-standard/features").StandardEventsOnMethod} */
+  #on(event, listener) {
+    this.#listeners[event].add(listener);
+    const off = () => {
+      this.#listeners[event].delete(listener);
+    };
+    return off.bind(this);
+  }
+
+  /**
+   *
+   * @param {WalletConnectAccount} account
+   * @returns {import("@wallet-standard/base").WalletAccount}
+   */
+  #walletConnectAccountToStandardAccount({ address }) {
+    const uint8Addr = base64ToUint8Ascii(address);
+
+    /** @type {import("@wallet-standard/base").IdentifierArray} */
+    // @ts-ignore
+    const features = Object.keys(this.features);
+    return {
+      address,
+      publicKey: uint8Addr,
+      chains: this.chains,
+      features,
+    };
+  }
+
+  /**
+   * @template {import("@wallet-standard/features").StandardEventsNames} E
+   * @param {E} event
+   * @param  {Parameters<import("@wallet-standard/features").StandardEventsListeners[E]>} args
+   */
+  #emit(event, ...args) {
+    for (const listener of this.#listeners[event]) {
+      // eslint-disable-next-line prefer-spread
+      listener.apply(null, args);
+    }
+  }
+
+  /**
+   * Defaults to first chain of this.chains if chain not provided
+   * @template {{ chain: NonLocalnetChain | null | undefined }} I
+   * @param {I[]} inputs
+   * @returns {Record<NonLocalnetChain, number[]>}
+   */
+  #inputsIndicesByChain(inputs) {
+    const { defaultChain } = this;
+    /** @type {Record<NonLocalnetChain, number[]>} */
+    const res = {
+      "solana:devnet": [],
+      "solana:mainnet": [],
+    };
+    for (let i = 0; i < inputs.length; i++) {
+      const { chain } = inputs[i];
+      if (chain) {
+        res[chain].push(i);
+      } else {
+        res[defaultChain].push(i);
+      }
+    }
+    return res;
+  }
+}
+
+/**
+ * @param {string} asciiStr base64 encoded string
+ * @returns {Uint8Array}
+ */
+function base64ToUint8Ascii(asciiStr) {
+  return Uint8Array.from(atob(asciiStr), (c) => c.charCodeAt(0));
+}
+
+/**
+ *
+ * @param {Uint8Array} asciiUint8Array
+ * @returns {string}
+ */
+function uint8ToBase64Ascii(asciiUint8Array) {
+  return btoa(String.fromCharCode(...asciiUint8Array));
+}
+
+/**
+ * @template {MaybeChainProp} I
+ * @param {I[]} inputs
```

Comment:
```suggestion
 * @param {readonly I[]} inputs
```

To fix type error 
```
Argument of type 'readonly SolanaSignTransactionInput[]' is not assignable to parameter of type 'SolanaSignTransactionInput[]'.
  The type 'readonly SolanaSignTransactionInput[]' is 'readonly' and cannot be assigned to the mutable type 'SolanaSignTransactionInput[]'
```
on line 259